src/runtime_src/xdp/profile/writer/native/native_writer.cpp

```diff
@@ -0,0 +1,91 @@
+/**
+ * Copyright (C) 2016-2021 Xilinx, Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may
+ * not use this file except in compliance with the License. A copy of the
+ * License is located at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+#define XDP_SOURCE
+
+#include "xdp/profile/writer/native/native_writer.h"
+#include "xdp/profile/database/database.h"
+
+#include "xdp/profile/plugin/vp_base/utility.h"
+
+namespace xdp {
+
+  NativeTraceWriter::NativeTraceWriter(const char* filename) :
+    VPTraceWriter(filename, "1.0", getCurrentDateTime(), 9 /* ns */)
+  {
+  }
+
+  NativeTraceWriter::~NativeTraceWriter()
+  {
+  }
+
+  void NativeTraceWriter::writeHeader()
+  {
+    VPTraceWriter::writeHeader() ;
+    fout << "XRT Version," << getToolVersion() << std::endl ;
+  }
+
+  void NativeTraceWriter::writeStructure()
+  {
+    // There is only one bucket where all the APIs will go
+    fout << "STRUCTURE" << std::endl ;
+    fout << "Group_Start,Host APIs" << std::endl ;
+    fout << "Group_Start,Native XRT API Calls" << std::endl ;
+    fout << "Dynamic_Row," << 1 << ",General,API Events" << std::endl ;
+    fout << "Group_End,Native XRT API Calls" << std::endl ;
+    fout << "Group_End,Host APIs" << std::endl ;
```

Comment:
std::endl incurs a stream flush, the cheapest is `<< '\n';` when flushing is not necessary.  Usually very last write should have the flush unless there is some file that is tail'ed by the user while it is being written.
https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rio-endl